modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,516 @@
+            log.error(customErrorMessage, xPathExpressionException);
+            throw new LazyLoadingTestException(customErrorMessage, xPathExpressionException);
+
+        } catch (IOException ioException) {
+            String customErrorMessage = "IOException exception  when login as tenant admin.";
+            log.error(customErrorMessage, ioException);
+            throw new LazyLoadingTestException(customErrorMessage, ioException);
+
+        } catch (SAXException saxException) {
+            String customErrorMessage = "SAXException exception  when login as tenant admin.";
+            log.error(customErrorMessage, saxException);
+            throw new LazyLoadingTestException(customErrorMessage, saxException);
+
+        } catch (XMLStreamException xmlStreamException) {
+            String customErrorMessage = "XMLStreamException exception  when login as tenant admin.";
+            log.error(customErrorMessage, xmlStreamException);
+            throw new LazyLoadingTestException(customErrorMessage, xmlStreamException);
+
+        } catch (LoginAuthenticationExceptionException loginAuthenticationExceptionException) {
+            String customErrorMessage = "LoginAuthenticationExceptionException exception  when login as tenant admin.";
+            log.error(customErrorMessage, loginAuthenticationExceptionException);
+            throw new LazyLoadingTestException(customErrorMessage, loginAuthenticationExceptionException);
+
+        } catch (URISyntaxException uriSyntaxException) {
+            String customErrorMessage = "URISyntaxException exception  when login as tenant admin.";
+            log.error(customErrorMessage, uriSyntaxException);
+            throw new LazyLoadingTestException(customErrorMessage, uriSyntaxException);
+
+        }
+
+
+    }
+
+    /**
+     * check the tenant is unloading functionality when tenant is idle more than configured tenant idle time.
+     * This method will wait additional MAX_THRESHOLD_TIME milliseconds to  system to unload the tenant. it will log
+     * total idle time taken to unload.
+     *
+     * @param tenantDomain - Tenant domain that need to check for unloading.
+     * @return boolean - true if tenant is unload after tenant idle time + additional MAX_THRESHOLD_TIME milliseconds if not return
+     * false
+     * @throws LazyLoadingTestException - Exception throws when  calling th REST call to get tenant in formation.
+     */
+    protected boolean checkTenantAutoUnloading(String tenantDomain) throws LazyLoadingTestException {
+        boolean isTenantUnloaded = false;
+        long totalSleepTime = 0;
+        log.info("Sleeping  for " + TENANT_IDLE_TIME + " milliseconds (Tenant idle tome).");
+
+        try {
+            Thread.sleep(TENANT_IDLE_TIME);
+        } catch (InterruptedException interruptedException) {
+            String customErrorMessage = "InterruptedException occurs when sleeping for TENANT_IDLE_TIME" +
```

Comment:
Add a comment explaining  the reason to not to throw the exception

comment:
Ignoring the InterruptedException because it will not impact to the main logic,
If InterruptedException occurs only impact it it will do the next loop before the expected sleep time.